### PR TITLE
Dedupe nested

### PR DESCRIPTION
**Pull request by RichieAHB**
~**Review this after the previous PR has been merged as this has branched from there and as such, contains those changes!** #383~

This change allows us to dedupe nested article fragments when dragging them from one front to another. Currently all that is happening here is that nested article fragments of dragged article fragments that already exist on the target front are being removed from the nested data of the drag and left in their original place in the front. This is most likely not ideal and the intention of the user is to probably replace the existing article fragments that are in the front and have them moved to this new position, but this is up for discussion!

### CODE REVIEW

File: client-v2/src/lib/guration/DedupeNode.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ...ExternalDedupeNodeProps,
+  context: DedupeContextType
+|};
+
+class DedupeNode extends React.Component<DedupeNodeProps> {
+  componentDidMount = () => this.reregister();
+  componentDidUpdate = () => this.reregister();
+  componentWillUnmount = () => this.deregister();
+
+  get dedupeContext() {
```
Comment (LATaylor-guardian) [inline client-v2/src/lib/guration/DedupeNode.js]: What is the `get` keyword doing here?

File: client-v2/src/lib/guration/README.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
+# Guration
```
Comment (LATaylor-guardian) [inline client-v2/src/lib/guration/README.md]: Great docs - nice work!

File: client-v2/src/lib/guration/DedupeNode.js (hunk 1 above)
Comment (RichieAHB) [reply client-v2/src/lib/guration/DedupeNode.js]: It means I can drop the parens when calling this method (as it has arity 0). Someone once told me about the "uniform access principle" and I ate it right up! Also it means I can destructure like below and it runs the getter. I only use them if they're not side effecting.

```js
const { dedupeContext } = this;
```

**Comment by RichieAHB** [comment]
@Reettaphant @jonathonherbert I've just tested this in the original fronts tool and it seems it doesn't care about nested stuff. You can have duplicates in nested articles ... is that a feature or just a limitation of the previous tool?

**Comment by RichieAHB** [comment]
This has been superseded

**Pull request closed without merging**